File: src/usage/providers/ollama.ts
/**
 * Ollama usage fetcher. Ollama does not expose a quota endpoint, so we emit
 * an empty report with an explanatory note — same shape as the Rust fetcher.
 */
import type { Database } from "bun:sqlite";
import type { UsageReport } from "../types.ts";
import type { Credential } from "../../auth/types.ts";
import type { UsageFetcher } from "../fetcher.ts";

export const ollamaFetcher: UsageFetcher = {
  provider: "ollama",
  async fetch(cred: Credential, _db: Database): Promise<UsageReport | null> {
    return {
      provider: "ollama",
      account: cred.type === "api_key" ? (cred.account ?? "api key") : (cred.email ?? cred.account_id ?? "account"),
      fetchedAt: Date.now(),
      limits: [],
      notes: ["Ollama does not expose a standalone quota usage API; per-response token usage is reported during requests."],
    };
  },
};

File: src/usage/providers/ollama.rs
use anyhow::Result;

use super::super::{UsageFetcher, UsageReport};
use crate::auth::Credential;

pub struct OllamaFetcher;

#[async_trait::async_trait]
impl UsageFetcher for OllamaFetcher {
    fn provider(&self) -> &'static str {
        "ollama"
    }

    async fn fetch(&self, cred: &Credential, _client: &reqwest::Client) -> Result<Option<UsageReport>> {
        Ok(Some(UsageReport {
            provider: "ollama".into(),
            account: cred.account_label(),
            fetched_at: chrono::Utc::now().timestamp_millis(),
            limits: Vec::new(),
            notes: vec![
                "Ollama does not expose a standalone quota usage API; per-response token usage is reported during requests.".into(),
            ],
        }))
    }
}
